modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/exp/ExpressionFactoryImpl.java

```diff
@@ -445,6 +500,52 @@ private ValueImpl(Scalar scalar, RowFactory<Row> factory) {
         }
     }
 
+    /** */
+    private class BiFieldGetter implements InputGetter {
+        /** */
+        private final Expression hnd_;
+
+        /** */
+        private final Expression row1_;
+
+        /** */
+        private final Expression row2_;
+
+        /** */
+        private final RelDataType rowType;
+
+        /** */
+        private BiFieldGetter(Expression hnd_, Expression row1_, Expression row2_, RelDataType rowType) {
+            this.hnd_ = hnd_;
+            this.row1_ = row1_;
+            this.row2_ = row2_;
+            this.rowType = rowType;
+        }
+
+        /** {@inheritDoc} */
+        @Override public Expression field(BlockBuilder list, int index, Type desiredType) {
+            Expression row1_ = list.append("row1", this.row1_);
+            Expression row2_ = list.append("row2", this.row2_);
+
+            Expression field = Expressions.call(hnd_,
+                IgniteMethod.ROW_HANDLER_BI_GET.method(),
+                Expressions.constant(index), row1_, row2_);
+
+            Type fieldType = typeFactory.getJavaClass(rowType.getFieldList().get(index).getType());
+
+            if (desiredType == null) {
+                desiredType = fieldType;
+                fieldType = Object.class;
+            } else if (fieldType != java.sql.Date.class
```

Comment:
Let's also reformat this block according to code-style rules.

Reply:
done, thanks !